src/bin/main.rs

```diff
@@ -1,24 +1,77 @@
 // use clap derive for argument parsing
 //
 // read a file name as the only argument
-use clap::Parser;
+use chrono::Local;
+use clap::{Arg, Parser};
+use env_logger::Builder;
+use log::{error, info, warn, LevelFilter};
+use std::io::Write;
+use uncrustable::eval;
 use uncrustable::parse::parse;
-
+use uncrustable::typecheck;
 #[derive(Parser, Debug)]
+
 struct Args {
     /// The input program to read
     input: String,
+
+    // The typecheck flag, can be used as --typecheck or --t
+    #[arg(short, long)]
+    typecheck: bool,
+
+    // The evaluate flag, can be used as --evaluate or --e
+    #[arg(short, long)]
+    evaluate: bool,
 }
 
 fn main() {
+    Builder::new()
```

Comment:
I don't see the purpose of this. You can just use `env_log` to log things without this initialization stuff. It's default prefix is already doing something like this.